src/ODM/MongoDB/Types/CurrencyType.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace ZFBrasil\DoctrineMoneyModule\ODM\MongoDB\Types;
+
+use Doctrine\ODM\MongoDB\Types\StringType;
+use Money\Currency;
+
+class CurrencyType extends StringType
+{
+    const NAME = 'currency';
+
+    public function getName()
+    {
+        return self::NAME;
+    }
+
+    public function convertToPHPValue($value)
+    {
+        if ($value === null || $value instanceof Currency) {
+            return $value;
+        }
+
+        $val = new Currency($value);
+
+        if (!$val) {
```

Comment:
Should probably check if `$value` is string before intantiation, then `return new Currency($value);`.